broadcast/dtos/dtos.go

```diff
@@ -0,0 +1,75 @@
+package dtos
+
+import (
+	"context"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"time"
+)
+
+type Msg interface {
+	GetBroadcastID() uint64
+	GetMethod() string
+	String() string
+}
+
+type BroadcastMsg struct {
```

Comment:
It would be quite useful with doc comments for structs, methods, fields etc. Especially, those fields that aren't common knowledge, such as `Info`, `Options`, `OriginAddr`.